test/unit/metadata_test.py

```diff
@@ -39,3 +42,60 @@ def test_get_etl_group_tables() -> None:
     """Test that a Value error is raised for non existent etl group."""
     with pytest.raises(ValueError):
         Package.get_etl_group_tables("not_an_etl_group")
+
+
+@pytest.mark.parametrize("resource_name", sorted(PUDL_RESOURCES.keys()))
+def test_pyarrow_schemas(resource_name: str):
+    """Verify that we can produce pyarrow schemas for all defined Resources."""
+    _ = PUDL_RESOURCES[resource_name].to_pyarrow()
```

Comment:
This was the main new test I wanted to add -- just so we catch any changes which happen to invalidate our PyArrow schemas (especially since we aren't actually using them yet, so it wouldn't break the builds)